jupyterlab_latex/api/api.yaml

```diff
@@ -16,12 +16,82 @@ paths:
           schema:
             type: string
             format: uri
+        - name: synctex
+          in: query
+          required: false
+          description: Whether to build the document using SyncTeX: 1 for true, and 0 for false.
+          schema:
+            type: integer
       responses:
         '200':
           description: The document was successfully built.
         '400':
           description: The request did not specify a .tex file.
-        '404':
+        '403':
           description: The request specified a file that does not exist.
         '500':
           description: The compilation steps for building the pdf had an error.
+  /latex/synctex/{filePath}:
+    get:
+      summary: Get a mapping between the text file and the compiled pdf.
+      parameters:
+        - name: filePath
+          in: path
+          required: true
+          description: The path of the .tex or .pdf file to map. If the path ends in .tex it gets the forward synchronization. If the path ends in .pdf it gets the revers synchronization.
+          schema:
+            type: string
+            format: uri
+        - name: line
+          in: query
+          required: false
+          description: The line of the text file for forwards synchronization.
+          schema:
+            type: integer
+        - name: column
+          in: query
+          required: false
+          description: The column of the text file for forwards synchronization.
+          schema:
+            type: integer
+        - name: page
+          in: query
+          required: false
+          description: The page of the PDF file for reverse synchronization.
+          schema:
+            type: integer
+        - name: x
+          in: query
+          required: false
+          description: The x position on the page (in pts) of the PDF file for reverse synchronization.
```

Comment:
how is this to be interpreted in the context of resizable windows?

Reply:
I'm not really sure. It comes from the SyncTex documentation, but I couldn't really experiment with it until figuring out how to get the same information on the PDF.js side. That's part of what I wanted to defer.

Reply:
From the perspective of SyncTex, points are dots, where pages are 72dpi. Not sure how to communicate that to PDF.js at the moment. It may be possible, but you know how the APIs are...